rfcs/convergence/react-focus-management.md

```diff
@@ -0,0 +1,182 @@
+
+Patterns in [ability-helpers](https://github.com/microsoft/ability-helpers) are **declarative**, and therefore decoupled more easily with framework and application code. Groups of focusables are managed by declaring `data-*` attributes which the lower level utilties use to define focusable behaviour.
+
+Nesting focusables is a problem that ability-helpers explicitly tries to solve, bringing focus management to a lower level than the rendering framework tries to accomplish this consistently independent of rendering cycles.
+
+### Modules
+
+The description of the each constituent module in the library can be found in the [ability-helpers README](https://github.com/microsoft/ability-helpers).
+
+The bundle size of each of the differnt modules of the library are listed in the below table. The library is fully tree shakeable, and for most cases only the `Core` part of the library is expected to be used
+
+| Module          | minified (kb) | gzipped(kb |
+|-----------------|---------------|------------|
+| Core            | 45.3          | 12.2       |
+| Deloser         | 13.2          | 3.2        |
+| Modalizer       | 4.5           | 0.8        |
+| ObservedElement | 3.1           | 0.7        |
+| Outline         | 7.9           | 2          |
+| CrossOrigin     | 18.9          | 4.4        |
+
+### Initial API
+
+#### FocusManagementProvider
+
+Context provider that hosts the [ability-helpers](https://github.com/microsoft/ability-helpers) instance and its API objects.
+
+The below table defines what is included in the context value.
+
+> Note that this table will increase as more ability-helpers features are included in Fluent
+
+| Name      | Value        | Description                           |
+| --------- | ------------ | ------------------------------------- |
+| focusable | FocusableAPI | API for traversing focusable elements |
+
+The below table defines the props used by the provider component. All context values can be used as props in the provider component but are optional.
+
+| Name                   | Value              | Description                                                                                                           |
+| ---------------------- | ------------------ | --------------------------------------------------------------------------------------------------------------------- |
+| dir                    | 'ltr' \| 'rtl'     | Direction indicator, some keyboard behaviour will need to change based on this value                                  |
+| Deloser                | DeloserAPI         | Tools to help automatically restore focus when it gets lost without manually calling `.focus()` from application code |
+| internal\_\_ahInstance | AbilityHelpersCore | The ability helpers instance, not expected to be be used directly                                                     |
```

Comment:
> not expected to be used directly.

I think having this in the description is precisely the reason why we might want to not expose this as we've had trouble in the past with people doing/using things "not expected" to be done/used.